### DISCUSSION TITLE
[feature request] disable border and margin when there is only one window in Tile mode

### DISCUSSION THREAD

**Discussion by Schroedingersdoraemon** [qtile/qtile#3014] [Ideas]
Hi, dear qtile developer.

I noticed that in the [Built-in Layouts](https://docs.qtile.org/en/latest/manual/ref/layouts.html#columns) part of the offcial doc the Columns mode has a key called "border on single", which draw the border when there is only one window. Margins and border width are useful but when there is only one window they seem less that useful.

So I am wondering if it is possible for you to add a key(or two respective keys) which can help to disable the margin and border width when there is only one window in Tile mode(and other modes if possible).

Best regards with appreciation

**Comment by m-col** [comment]
> So I am wondering if it is possible for you to add a key(or two respective keys) which can help to disable the margin and border width when there is only one window in Tile mode(and other modes if possible).

Do you mean when using the `Tile` layout? Adding a `border_on_single` option to the `Tile` layout would be pretty straight forward - perhaps you'd like to have a try at implementing it?

As for the margins, something similar would need to be implemented on the layouts which I think would have similar logic in the `configure` methods:
```py
if self.margin_on_single and len(self.clients) == 1:
    margin = 0
else:
    margin = self.margin
```

**Reply by Schroedingersdoraemon** [reply]
Dear developer

Thanks for your reply, I have read your advice and tried to modify the source code.  Actually I successfully achieved the border_on_single and margin_on_single in Tile mode.  Below are the details.

You see, the border_on_single defined in func configure() works well.  Margin_on_single, however, fails to work if similarly defined in func configure().  But if defined as the parameter of func client.place(), margin_on_single could work.  

If you think the following changes could satisfy the requirements I will then submit a new PR.

```patch
50a51
>         ("border_on_single", 0, "Draw border if there is only one window"),
52a54
>         ("margin_on_single", 0, "Draw margin if there is only one window"),
135a138,139
>         # border_on_single = self.border_on_single    #  I wonder if these two lines
>         # margin_on_single = self.margin_on_single    #  should be kept ?
153a158,161
>             if len(self.clients)==1 and not self.border_on_single:
>                 border_width = 0
>             else:
>                 border_width = self.border_width
161c169
<                 margin=self.margin,
---
>                 margin = 0 if (len(self.clients)==1 and not self.margin_on_single) else self.margin
```

Best regards

**Comment by Schroedingersdoraemon** [comment]
#3020

I implemented this function in this PR, which is being waited to be merged to master
